File: Builder/PHP/PDFReportBuilder.php
<?php

require_once "IReportBuilder.php";
require_once "Report.php";

class PDFReportBuilder implements IReportBuilder {
    private Report $report;

    public function buildContent()
    {
        if (!isset($this->report)) {
            $this->report = new Report;
        }

        $this->report->setContent("Summary");
    }

    public function buildFile()
    {
        if (!isset($this->report)) {
            $this->report = new Report;
        }

        $this->report->setFilename("Report.PDF");
    }

    public function getResult() {
        return $this->report;
    }
}

File: Builder/Python 3/PDFReportBuilder.py
from IReportBuilder import IReportBuilder
from Report import Report

class PDFReportBuilder(IReportBuilder):
  def __init__(self) -> None:
      super().__init__()

  def buildContent(self):
    if not self._report:
      self._report = Report()
    
    self._report.content = "Summary"

  def buildFile(self):
    if not self._report:
      self._report = Report()
    
    self._report.filename = "Report.pdf"

  def getResult(self):
    return self._report